### PR TITLE
Fake Data Generator

### PR DESCRIPTION
**Pull request by Jen-Hall**
# Generating fake data

## Overview and Purpose
 The purpose of `generate_fake_data.py` is to create a `.csv` file with fake data with the following intended applications:
- An example of how data needs to be formatted to be passed into the model
- To ensure the files are being generated correctly to test GUI setup.
- To test the setup and running of the repo

The purpose of the fake data is *NOT* too:
- *DO NOT* use the model artifacts (`.pickle` and `.state` files) generated from the fake data to make predictions which will be used in any real world application.
- *DO NOT* use the fake data generated to inform any insights to be applied to a real world setting.
- *DO NOT* use to test the performance of the model

The data is generated completely randomly, with each field having random values generated independently of other fields. This generator was created having never been exposed to the real data.

### CODE REVIEW

File: fake_data_generation/generate_fake_data.py [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+)
+
+parser.add_argument(
+    "--only_major_cases",
+    "-mc",
+    default=False,
+    action="store_true",
+    help=""" [False - no need to specify, True - specify by just including: --only_major_cases]
+    If True all records generated will have major cases listed as 'Y' if False cases will be a mix of 'N' and 'Y'.""",
+)
```
[the remark this answers is not in the document]
Comment (Jen-Hall) [reply fake_data_generation/generate_fake_data.py:45]: Thank you for highlighting - I will update.

**Review by matthewcooper19 — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -52,6 +52,7 @@ stratified those patients with long and serious hospital stays under the long-st
 | [Configuration Files](config/README.md) | Overview of provided configuration files |
 | [Production Build Configuration Files](deploy/README.md) | Overview of the configuration files provided for production build Docker containers |
 | [Training](training/README.md) | Description of the training process for the models used in the LTSS API |
+| [Generating fake data](fake_data_generation/README.md) | Description of how to generate fake data to test the setup and running of the repo |
```
Comment (amadeuspzs) [inline README.md:55]: I would move this higher in the table as logically this will come before testing

File: fake_data_generation/README.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
+# Generating fake data
+
+## Overview and Purpose
+This directory contains a file called `generate_fake_data.py`. The purpose of `generate_fake_data.py` is to create a `.csv` file with fake data with the following intended applications:
+- An example of how data needs to be formatted to be passed into the model
+- To ensure the files are being generated correctly to test GUI setup.
```
Comment (amadeuspzs) [inline fake_data_generation/README.md:6]: Remove . at end of bullet or add to other bullets for consistency

File: fake_data_generation/README.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
+# Generating fake data
+
+## Overview and Purpose
+This directory contains a file called `generate_fake_data.py`. The purpose of `generate_fake_data.py` is to create a `.csv` file with fake data with the following intended applications:
+- An example of how data needs to be formatted to be passed into the model
```
Comment (amadeuspzs) [inline fake_data_generation/README.md:5]: `As an example`

File: fake_data_generation/README.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+- To ensure the files are being generated correctly to test GUI setup.
+- To test the setup and running of the repo
+
+The following should be noted with regards to artifacts produced by using fake data:
+- *DO NOT* use the model artifacts (`.pickle` and `.state` files) generated from the fake data to make predictions which will be used in any real world application.
+- *DO NOT* use the fake data generated to inform any insights to be applied to a real world setting.
+- *DO NOT* use to test the performance of the model
```
Comment (amadeuspzs) [inline fake_data_generation/README.md:12]: Add . at end or remove from other bullets.

File: fake_data_generation/README.md [hunk 6]
```diff
@@ -0,0 +1,58 @@
+# Generating fake data
+
+## Overview and Purpose
+This directory contains a file called `generate_fake_data.py`. The purpose of `generate_fake_data.py` is to create a `.csv` file with fake data with the following intended applications:
+- An example of how data needs to be formatted to be passed into the model
+- To ensure the files are being generated correctly to test GUI setup.
+- To test the setup and running of the repo
+
+The following should be noted with regards to artifacts produced by using fake data:
+- *DO NOT* use the model artifacts (`.pickle` and `.state` files) generated from the fake data to make predictions which will be used in any real world application.
```
Comment (amadeuspzs) [inline fake_data_generation/README.md:10]: `artefact` is British spelling

File: fake_data_generation/README.md (hunk 5 above)
Comment (amadeuspzs) [reply fake_data_generation/README.md:12]: Missing noun: `the fake data` ?

File: fake_data_generation/README.md [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+
+The data is generated completely randomly, with each field having random values generated independently of other fields. This generator was created having never been exposed to the real data.
+
+The categories for the fields in the fake data can be found in [fake_data_categories.json](../config/fake_data_categories.json). The categories specified in this file are randomly chosen to fill the corresponding fields in the fake data. This is done in `generate_fake_data.py`.
+
+Note a number of the fields will show only test categories and may not be reflective of the types of categories or number of categories types in each field. However the formatting of the fields is reflective as an example of what is required to run the model.
+
+Data fields that do not have their categories specified in [fake_data_categories.json](../config/fake_data_categories.json) or is not a categorical variable has the fake data required generated in `generate_fake_data.py` line by line to show how the data is being generated for each field. In `generate_fake_data.py` the fields are split into the field data type (e.g. str, int) required for the models to train.
```
Comment (amadeuspzs) [inline fake_data_generation/README.md:20]: `or are not categorical variables have the`

File: fake_data_generation/README.md [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+$ python3 generate_fake_data.py -nr 200 -fn "training_data" --only_major_cases
+```
+or depending on your machine setup:
+
+  ```bash
+$ python generate_fake_data.py -nr 200 -fn "training_data" --only_major_cases
+```
+
+## Generating fake data to test repo
+For a step by step guide on how to generate the fake data and then train the models to test the repo setup. Please see: [Training](../training/README.md)
```
Comment (amadeuspzs) [inline fake_data_generation/README.md:58]: `,please see`

File: training/README.md [hunk 9]
```diff
@@ -124,5 +126,80 @@ optional arguments:
...
+```
+$ python3 train_los.py -d 'fake_training_data.csv' -e 2 --shuffle-data --shuffle-seed 10 --save-frequency 1
+```
+
+Note the data the model with this command is not being trained to optimise performance for prediction. The settings in the command above are so the training is completed faster to generate a file that can be used to test the set up.
+
+## 4. Saving the LoS model as `.state`
+This is to save the trained LoS model.
+
+In the `training` you should now see three files `mod_ep_0`, `mod_ep_1`, `mod_ep_2`. For the purposes of just converting a file to test the set up of the repo, `mod_ep_2` will be renamed to `los_model.state` .
```
Comment (amadeuspzs) [inline training/README.md:188]: In the `training` folder

File: training/README.md [hunk 10]
```diff
@@ -124,5 +126,80 @@ optional arguments:
...
+This is to run the risk model.
+
+Run the following command (remaining in the same directory: `training`).
+
+```bash
+$ python3 train_risk.py -d 'fake_training_data.csv' -s risk_model.pickle --shuffle-data --shuffle-seed 100
 ```
+A file called `risk_model.pickle` should now appear in the `training` directory.
+
+All the files have now been generated to test the set up. Instructions on deploy can be [found here](../Deploy/README.md).
```
Comment (amadeuspzs) [inline training/README.md:205]: You will need to copy these to the `config` folder before following instructions on how to deploy [here]

**Review by amadeuspzs — CHANGES_REQUESTED** [review]
Great work @Jen-Hall; I can confirm running the scripts and deploying works.

**Review by amadeuspzs — APPROVED** [review]
Closes #5

**Pull request merged**